### PR TITLE
Simplify argument handling

### PR DESCRIPTION
**Pull request by islemaster**
We were handling function arguments in a needlessly complex way in a few cases.  Explanations inline.

### CODE REVIEW

File: lib/p5.play.js [hunk 1]
```diff
@@ -234,30 +234,15 @@ p5.prototype.getSprites = function() {
...
-  {
-    if(arguments[0] instanceof Array === false)
-      throw("Error: with drawSprites you can only draw all sprites or a group");
-    else
-    {
-      arguments[0].draw();
-    }
 
-  }
+  group.draw();
```
Comment (islemaster) [inline lib/p5.play.js]: I carefully compared the implementations of `group.draw()` and the sort/display work being done on `allSprites` here, and discovered that they're identical.  Therefore, the whole implementation of this method collapses into a few lines.

File: lib/p5.play.js [hunk 2]
```diff
@@ -1820,8 +1805,7 @@ function Sprite(pInst, _x, _y, _w, _h) {
   * @return {Boolean} result True if non-transparent
   */
   this.overlapPixel = function(pointX, pointY) {
-
-    var point = createVector(arguments[0], arguments[1]);
+    var point = createVector(pointX, pointY);
```
Comment (islemaster) [inline lib/p5.play.js]: There was no reason to get the named arguments `pointX` and `pointY` by indexing into `arguments` here.

File: lib/p5.play.js [hunk 3]
```diff
@@ -1855,7 +1839,7 @@ function Sprite(pInst, _x, _y, _w, _h) {
   * @return {Boolean} result True if inside
   */
   this.overlapPoint = function(pointX, pointY) {
-    var point = createVector(arguments[0], arguments[1]);
+    var point = createVector(pointX, pointY);
```
Comment (islemaster) [inline lib/p5.play.js]: Likewise re: named arguments here.

File: lib/p5.play.js [hunk 4]
```diff
@@ -3413,21 +3397,18 @@ function Animation(pInst) {
...
+   *
+   * @method draw
+   * @param {Number} x x coordinate
+   * @param {Number} y y coordinate
+   * @param {Number} [r=0] rotation
+   */
+  this.draw = function (x, y, r) {
+    this.xpos = x;
+    this.ypos = y;
+    this.rotation = r || 0;
```
Comment (islemaster) [inline lib/p5.play.js]: `Animation.draw()` had an undocumented optional third argument.  I've documented it, simplified its use, and replaced the unnecessary uses of `arguments` with named parameters.

**Comment by toolness** [comment]
Nice, this is great! Thanks.

**Pull request merged**